src/contracts/atlas/Escrow.sol

```diff
@@ -313,6 +306,37 @@ abstract contract Escrow is AtlETH {
         return (result, gasLimit);
     }
 
+    /// @notice Validates a SolverOperation's deadline against the current block.
+    /// @param solverOp The SolverOperation being validated.
+    /// @param dConfig DApp configuration data, including solver gas limits and operation parameters.
+    /// @return result Updated result flags after performing the validation checks, including any new errors
+    function _validateSolverOpDeadline(
+        SolverOperation calldata solverOp,
+        DAppConfig memory dConfig
+    )
+        internal
+        view
+        returns (uint256 result)
+    {
+        if (solverOp.deadline != 0 && block.number > solverOp.deadline) {
```

Comment:
this same check still happens in `_validateSolverOpGas()`?

Reply:
removed the check